skyportal/handlers/api/spectrum.py

```diff
@@ -154,7 +157,10 @@ def get(self, spectrum_id):
         """
 
         spectrum = Spectrum.get_if_accessible_by(
-            spectrum_id, self.current_user, raise_if_none=True
+            spectrum_id,
+            self.current_user,
+            raise_if_none=True,
+            options=[joinedload(Spectrum.comments)],
```

Comment:
Rather than a joinedload, which loads every related record while ignoring access permissioning, you should load these manually w/ another query_records_accessible_by after fetching the spectrum